### ISSUE TITLE
Controlled tabs focus marker not working when changing tabs delayed

### ISSUE THREAD

**Issue by MiK546** [jamesmfriedman/rmwc#442]
### Environment
RMWC 5.2.1, Firefox/Chrome

### Problem
When using controlled tabs and delaying changing the active tab using a dialog, the visibly focused tab doesn't change correctly. It remains as the previously active tab. The new tab must be pressed again for the focus marker to change correctly.

### Example code
```js
import React from "react";
import ReactDOM from "react-dom";
import { TabBar, Tab } from "@rmwc/tabs";
import { SimpleDialog } from "@rmwc/dialog";

import "@material/dialog/dist/mdc.dialog.css";
import "@material/button/dist/mdc.button.css";
import "@material/tab-bar/dist/mdc.tab-bar.css";
import "@material/tab/dist/mdc.tab.css";
import "@material/tab-scroller/dist/mdc.tab-scroller.css";
import "@material/tab-indicator/dist/mdc.tab-indicator.css";

class App extends React.Component {
  constructor(props) {
    super(props);

    this.newTab = null;
    this.state = {
      /** The index of the currently active tab. */
      tabIndex: 0,
      /** True if the chnage tabs dialog is open. */
      changingTabs: false
    };
  }

  render() {
    return (
      <React.Fragment>
        <TabBar
          activeTabIndex={this.state.tabIndex}
          onActivate={tab => {
            // Store the new tab index for later
            this.newTab = tab.detail.index;
            // Open the dialog
            this.setState({
              changingTabs: true
            });
          }}
        >
          <Tab>Cookies</Tab>
          <Tab>Pizza</Tab>
          <Tab>Icecream</Tab>
        </TabBar>

        Selected tab: {this.state.tabIndex}

        <SimpleDialog
          title={"Switch tabs?"}
          open={this.state.changingTabs}
          acceptLabel={"Really change"}
          cancelLabel={"Don't change"}
          onClose={evt => {
            // Close the dialog
            let stateChanges = {
              changingTabs: false
            };
            // If the user pressed accept, change tabs
            if (evt.detail.action === "accept") {
              stateChanges.tabIndex = this.newTab;
            }
            this.setState(stateChanges);
          }}
        />
      </React.Fragment>
    );
  }
}

const rootElement = document.getElementById("root");
ReactDOM.render(<App />, rootElement);
```
https://codesandbox.io/s/982rjv619y

In the example, a dialog pops up when trying to change a tab. If the user selects "Really change", the tab is changed. The `state.tabIndex` changes correctly. When looking at the tab bar however, visually the previous tab is still active.

**Comment by jamesmfriedman** [comment]
Thanks for reporting this. Not sure whats special about controlled tabs in this specific situation but it may have to do with the dialog stealing focus. I'll look into it

**Comment by jamesmfriedman** [comment]
fixed in 5.4.0

After fixing the bug, it exposed some errors in the logic of the example. Here is some working TS code

```
class Bug442 extends React.Component {
  newTab: any = null;

  state = {
    /** The index of the currently active tab. */
    tabIndex: 0,
    /** True if the chnage tabs dialog is open. */
    changingTabs: false
  };

  render() {
    return (
      <React.Fragment>
        <TabBar
          activeTabIndex={this.state.tabIndex}
          onActivate={tab => {
            // Store the new tab index for later
            if (this.newTab !== tab.detail.index) {
              this.newTab = tab.detail.index;
              // Open the dialog
              this.setState({
                changingTabs: true
              });
            }
          }}
        >
          <Tab>Cookies</Tab>
          <Tab>Pizza</Tab>
          <Tab>Icecream</Tab>
        </TabBar>
        Selected tab: {this.state.tabIndex}
        <SimpleDialog
          title={'Switch tabs?'}
          open={this.state.changingTabs}
          acceptLabel={'Really change'}
          cancelLabel={"Don't change"}
          onClose={evt => {
            // Close the dialog
            let stateChanges: any = {
              changingTabs: false
            };
            // If the user pressed accept, change tabs
            if (evt.detail.action === 'accept') {
              stateChanges.tabIndex = this.newTab;
            } else {
              this.newTab = null;
            }

            this.setState(stateChanges);
          }}
        />
      </React.Fragment>
    );
  }
}
```
